### ISSUE TITLE
Lost all my rules 

### ISSUE THREAD

**Issue by akasaka99** [kintesh/containerise#152]
Hi-
After a sudden crash of FF 85 on MacOs (13.6) , I lost all my Container Rules and could not find them even though i used FF sync.
However, I make regular disk backups so I presume I could recover those rules somewhere.
Could someone tell me which are the file(s) that I wuld need to use to recover my rules.
Many thanks !

**Comment by michaelrigart** [comment]
@akasaka99 : I encountered the same issue 3 days ago ( macOS 11.2.1 BigSur) .  But I was unable to recover the rules so I had to reconfigure then. I'm now keeping a separate csv file just in case this might happen again.

**Comment by akasaka99** [comment]
Yes I had to it all over again, its a real pain. I found your could recover the containers in your FF profile in the file containers.json but only the containers not the rules.
Do you mind explaining:
1) What format/column you use for your cvs file: [url, container_name] I presume?
2) Where do you store this cvs file aftwerards (in FF profile) when you want for instance to reinstate back all those containers rules. And you know which name it should be stored under? 
Or maybe you have some doc as I cant find anything to reinstate those containers rule again. 
Thanks !

**Comment by michaelrigart** [comment]
@akasaka99 

1. Yes, de csv I'm keeping as a backup is indeed comma separated with url, container_name.
2. I keep the csv file somewhere in my documents folder, not in the FF profile. This way it is present in my own backups.

**Comment by akasaka99** [comment]
Yes indeed.
And in case there's another problem with FF, how do you import those data that you saved back into your containers?
Or you just re-enter everything manually?

**Comment by michaelrigart** [comment]
No, I open the csv file in a text editor and just copy-paste the content of my csv-file into the csv-editor of Containerise 

![Screenshot 2021-03-02 at 11 06 01](https://user-images.githubusercontent.com/242477/109632459-5aba8600-7b47-11eb-8d73-2daee9a3a7a0.png)

**Comment by akasaka99** [comment]
Wow, I learned something, I had not clue there was a containerise cvs editor.
I just looked into FF about:addons and I see the only  General Preferences and Containers Preferences   but I cant see containerise cvs editor. Where is the editor pls?

**Comment by michaelrigart** [comment]
You can open the plugin from your plugin-toolbar. On the right-side, you have the little pencil icon ;) . Just click on it and it will open the csv-editor.

![Screenshot 2021-03-02 at 11 31 37](https://user-images.githubusercontent.com/242477/109635664-eda8ef80-7b4a-11eb-89bf-1497c0b39532.png)

**Comment by akasaka99** [comment]
Excellent. Many thanks for your help !!!
Have a great day

**Comment by designt** [comment]
Well I need a backup the csv file, what's the path to that file ?
Thanks
